### PR TITLE
CAM-12337 adjust springboot examples for 7.14 release

### PR DESCRIPTION
**Pull request by mboskamp**

### CODE REVIEW

File: spring-boot-starter/example-autodeployment/pom.xml [hunk 1]
```diff
@@ -7,8 +7,8 @@
   <version>0.0.1-SNAPSHOT</version>
 
   <properties>
-    <spring.boot.version>2.2.5.RELEASE</spring.boot.version>
-    <camunda.spring.boot.starter.version>7.13.0</camunda.spring.boot.starter.version>
+    <spring.boot.version>2.3.1.RELEASE</spring.boot.version>
```
Comment (tmetzke) [inline spring-boot-starter/example-autodeployment/pom.xml:10]: Spring Boot 2.3.4.RELEASE is out, let's use it if it works (should work fine but please test it shortly 😃)
```suggestion
    <spring.boot.version>2.3.4.RELEASE</spring.boot.version>
```

File: spring-boot-starter/example-simple/pom.xml [hunk 2]
```diff
@@ -57,14 +57,14 @@
...
-      <version>4.11.0</version>
+      <version>4.13.0</version>
       <scope>test</scope>
     </dependency>
 
     <dependency>
       <groupId>org.camunda.bpm.assert</groupId>
       <artifactId>camunda-bpm-assert</artifactId>
-      <version>6.0.0</version>
+      <version>7.0.0</version>
```
Comment (tmetzke) [inline spring-boot-starter/example-simple/pom.xml:67]: let's use 8.0.0 already (not released yet but will be for 7.14)

File: spring-boot-starter/example-autodeployment/README.md [hunk 3]
```diff
@@ -36,7 +36,7 @@ public class AutoDeploymentApplication {
 
 3. You can also put BPMN, CMMN and DMN files in your classpath, they will be automatically deployed and registered within the process application.
 
-4. When implementing the process application in [standard manner](https://docs.camunda.org/manual/7.13/user-guide/process-applications/the-process-application-class/),
+4. When implementing the process application in [standard manner](https://docs.camunda.org/manual/7.14/user-guide/process-applications/the-process-application-class/),
```
Comment (tmetzke) [inline spring-boot-starter/example-autodeployment/README.md:39]: Should we simply go for `latest` here so it doesn't have to be upgraded every time?

File: spring-boot-starter/example-invoice/pom.xml [hunk 4]
```diff
@@ -7,8 +7,8 @@
   <version>0.0.1-SNAPSHOT</version>
```
Comment (tmetzke) [inline spring-boot-starter/example-invoice/pom.xml:7]: Let's update `jaxb-impl` to `2.2.4` as well

File: spring-boot-starter/example-twitter/README.md [hunk 5]
```diff
@@ -40,7 +40,7 @@ public class TwitterServletProcessApplication {
 
 3. You can also put BPMN, CMMN and DMN files in your classpath, they will be automatically deployed and registered within the process application.
 
-4. You can configure your Spring Boot application using `application.yaml` file. All possible Camunda-specific configuration parameters are listed [here](https://docs.camunda.org/manual/7.13/user-guide/spring-boot-integration/configuration/)
+4. You can configure your Spring Boot application using `application.yaml` file. All possible Camunda-specific configuration parameters are listed [here](https://docs.camunda.org/manual/7.14/user-guide/spring-boot-integration/configuration/)
```
Comment (tmetzke) [inline spring-boot-starter/example-twitter/README.md:43]: Should we simply go for `latest` here so it doesn't have to be upgraded every time?

File: spring-boot-starter/example-web/README.md [hunk 6]
```diff
@@ -38,7 +38,7 @@ public class RestApplication {
 
 3. You can also put BPMN, CMMN and DMN files in your classpath, they will be automatically deployed and registered within a process application.
 
->Note: Basic authentication is configured on Spring Boot level, this is NOT a Camunda Engine authentication. Check [this docs](https://docs.camunda.org/manual/7.13/reference/rest/overview/authentication/) 
+>Note: Basic authentication is configured on Spring Boot level, this is NOT a Camunda Engine authentication. Check [this docs](https://docs.camunda.org/manual/7.14/reference/rest/overview/authentication/)
```
Comment (tmetzke) [inline spring-boot-starter/example-web/README.md:41]: Should we simply go for `latest` here so it doesn't have to be upgraded every time?

**Review by tmetzke — APPROVED** [review]
Looks good overall, 👍 
Let's bump some versions here even further and maybe use the `latest` branch of the docs?

**Comment by tmetzke** [comment]
Also, can we relate the whole thing rather to CAM-12344? :)

**Comment by mboskamp** [comment]
Regarding `latest` instead of specific versions: I had the same discussion for 7.13 and we settled for using the version because the examples are meant to be checked out. Using the latest docs might lead to misinforming the user.

**Review by tmetzke — APPROVED** [review]
Looks good 👍

**Pull request merged**